src/status_im/common/controlled_input/utils.cljs

```diff
@@ -0,0 +1,98 @@
+(ns status-im.common.controlled-input.utils
+  (:require
+    [clojure.string :as string]
+    [reagent.core :as reagent]))
+
+(defn create-input-state
+  []
+  (reagent/atom {:value       ""
+                 :error?      false
+                 :upper-limit nil}))
+
+(defn input-value
+  [state]
```

Comment:
Maybe we can pass the de ref value for all these functions instead ?

Reply:
My idea was to hide the implementation from UI, so the components wouldn't even bother what is behind that state. I don't mind dereferencing outside but not sure what would be the benefit?

Reply:
it just means the passing function doesn't have to aware of the type being atom.